### PR TITLE
task 10 done

### PR DESCRIPTION
**Pull request by MolchanovaA**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -278,3 +278,18 @@ describe("task 7. POST /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe.only("Task 10 GET /api/users", () => {
```
Comment (nc-verity-gregory) [inline __tests__/app.test.js:281]: Please don't leave .only in.

**Review by nc-verity-gregory — CHANGES_REQUESTED** [review]
I can't review this for you at the moment Alina, with .only in the test that circumvents the pre commit hook. Please remove it and resubmit.

File: controllers/news.controller.js [hunk 2]
```diff
@@ -106,3 +105,17 @@ exports.postNewComment = (req, res, next) => {
...
+    data.forEach(item =>{
+      if(!item.username || !item.name || !item.avatar_url){
+        return Promise.reject({code: 404, msg: 'not found'})
+      }
+    })
+    res.status(200).send({ [table]: data });
+  }).catch(err=>{
+    next(err)
+  });
+};
```
Comment (nc-mick-fay) [inline controllers/news.controller.js:121]: I really like the changes you have made to make your receiveAll function totally reusable by just passing in the table name. Really nice solution!

File: controllers/news.controller.js [hunk 3]
```diff
@@ -106,3 +105,17 @@ exports.postNewComment = (req, res, next) => {
...
     .catch(next);
 };
+
+exports.getAllUsers = (req, res) => {
+  const table = "users";
+  receiveAll(table).then((data) => {
+    data.forEach(item =>{
+      if(!item.username || !item.name || !item.avatar_url){
+        return Promise.reject({code: 404, msg: 'not found'})
+      }
```
Comment (nc-mick-fay) [inline controllers/news.controller.js:115]: As your database contains columns of all these values, the keys should always exist on the data. I don't think you need to add this check in here, as you're checking in the test suite.

This check would also mean if someone inserted a user without an ID, your endpoint would throw an error, which may not be the functionality you want.

**Review by nc-mick-fay — APPROVED** [review]
Good work on this endpoint. I really like the totally reusable `receiveAll` model you've created - it saves you from making lots of models that basically do the same thing

I'm happy to approve this pull request, merge into main when you're ready!

Something to consider - in your controller you have a check you if the received data has a username/name/avatarurl property. If there was a user created without an avatar, would you want this endpoint to then stop working? It doesn't need changing, as it's your code and you decide the functionality you want!

**Pull request merged**